luamocks/LuaSyncedCtrl.lua

```diff
@@ -19,202 +19,441 @@ stringMock ="TestString"
 tableMock ={exampletable= true}
 booleanMock =true
 functionMock =function (bar) return bar; end
+
+--==================================================================================================
+-- GameRulesParameter
 --==================================================================================================
+--[[ 
+---@param paramName string
+---@param losAccess? losAccess
+---@return string
+function Spring.SetGameRulesParam   (  paramName, losAccess )
+	assert(type(paramName) == "string","Argument paramName is of invalid type - expected string");
+	assert(losAccess == "private" | losAccess == "allied" | losAccess == "inlos" | losAccess == "inradar"
+		| losAccess == "public", "Argument losAccess is invalid");
+return  stringMock
+end
 
-function Spring.SetUnitRulesParam (  unitID, paramName)
-assert(type(unitID) == "number","Argument unitID is of invalid type - expected number");
-assert(type(paramName) == "string","Argument paramName is of invalid type - expected string");
-return  numberMock
- end
+---@param teamID number
+---@param paramName string
+---@param losAccess? losAccess
+---@return integer
+function Spring.SetTeamRulesParam (teamID, paramName, losAccess)
+	assert(type(teamID) == "number","Argument teamID is of invalid type - expected number");
+	assert(type(paramName) == "string" or type(paramName) == "number","Argument paramName is of invalid type - expected string or number");
+	assert(losAccess == "private" | losAccess == "allied" | losAccess == "inlos" | losAccess == "inradar"
+		| losAccess == "public", "Argument losAccess is invalid");
```

Comment:
param must be a string. it is the value that can be a number. los access can be a table, plus inlos/inradar aren't really meaningful for teams

Reply:
![image](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/806377/644006a0-89a0-44c1-966b-7501617eb331)
See here for what I intend to have currently, I adjusted it based on your comment here (in the other file).